pkg/auth/cache/metrics_accesscache_test.go

```diff
@@ -0,0 +1,227 @@
+package cache_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"go.uber.org/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/konflux-ci/namespace-lister/pkg/auth/cache"
+	"github.com/konflux-ci/namespace-lister/pkg/auth/cache/mocks"
+)
+
+const (
+	resourcesRequestsMetricFullname     = "namespace_lister_accesscache_resource_requests_total"
+	timeRequestsMetricFullname          = "namespace_lister_accesscache_time_requests_total"
+	syncMetricFullname                  = "namespace_lister_accesscache_synch_op_total"
+	subjectNamespacePairsMetricFullname = "namespace_lister_accesscache_subject_namespace_pairs"
+	subjectsMetricFullname              = "namespace_lister_accesscache_subjects"
+
+	metadataSynchOp = `# HELP namespace_lister_accesscache_synch_op_total synchronization operations
```

Comment:
We should maybe use [prometheus's common library](https://pkg.go.dev/github.com/prometheus/common@v0.62.0/expfmt) to parse this data, rather than doing string comparisons.  We can do that in a follow-up PR if you want.

Reply:
Good ideas. Let's look into that on a follow up PR